core/include/detray/geometry/surfaces_finder.hpp

```diff
@@ -0,0 +1,137 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "detray/definitions/detail/accessor.hpp"
+#include "detray/grids/axis.hpp"
+#include "detray/grids/grid2.hpp"
+#include "detray/grids/populator.hpp"
+#include "detray/grids/serializer2.hpp"
+#include "detray/tools/grid_array_helper.hpp"
+
+#pragma once
+
+namespace detray {
+
+template <std::size_t N,
+          template <typename, unsigned int> class array_type = darray,
+          template <typename...> class tuple_type = dtuple,
+          template <typename...> class vector_type = dvector,
+          template <typename...> class jagged_vector_type = djagged_vector>
+struct surfaces_finder {
+    public:
+    surfaces_finder() = delete;
+
+    using surfaces_regular_circular_grid =
+        grid2<attach_populator, axis::regular, axis::circular, serializer2,
+              vector_type, jagged_vector_type, array_type, tuple_type, dindex,
+              false>;
+
+    static const size_t N_GRIDS = N;
+
+    /// host constructor
+    template <std::size_t... ints>
+    DETRAY_HOST surfaces_finder(vecmem::memory_resource& resource)
+        : _surface_grids(host(resource, std::make_index_sequence<N_GRIDS>{})) {}
+
+    template <std::size_t... ints>
+    DETRAY_HOST auto host(vecmem::memory_resource& resource,
```

Comment:
Can we rename this to something more descriptive or at least put comments? Maybe I am missing something, but why is this separate from the constructor?

Reply:
I will rename `host` and `device` properly .

I separate them to skip `std::index_sequence`  argument in constructor.
It's going to be nice to have everything in constructor with default argument:
```c++
    /// host constructor
    template <std::size_t... ints>
    DETRAY_HOST surfaces_finder(vecmem::memory_resource& resource, 
                                std::index_sequence<ints...>=std::make_index_sequence<N_GRIDS>{})
        : _surface_grids(...) {}
```

However, in this case, `< std::size_t... ints > ` is not deductible... that's why there is a separate function.